src/librustc/ty/query/caches.rs

```diff
@@ -21,9 +21,9 @@ pub(crate) trait QueryCache: Default {
     /// It returns the shard index and a lock guard to the shard,
     /// which will be used if the query is not in the cache and we need
     /// to compute it.
-    fn lookup<'tcx, R, GetCache, OnHit, OnMiss>(
+    fn lookup<R, GetCache, OnHit, OnMiss>(
         &self,
-        state: &'tcx QueryState<TyCtxt<'tcx>, Self>,
+        state: &QueryState<CTX, Self>,
```

Comment:
Can't you just have a `CTX: QueryContext` parameter on the `lookup` and `complete` methods instead?